main/src/main/java/cgeo/geocaching/utils/ClipboardUtils.java

```diff
@@ -37,10 +37,18 @@ public static void copyToClipboard(@NonNull final CharSequence text) {
     @Nullable
     public static String getText() {
         final ClipboardManager clipboard = (ClipboardManager) CgeoApplication.getInstance().getSystemService(Context.CLIPBOARD_SERVICE);
-        final ClipData clip = clipboard.getPrimaryClip();
-        if (clip != null && clip.getItemCount() > 0) {
-            final CharSequence text = clip.getItemAt(0).getText();
-            return text != null ? text.toString() : null;
+        if (clipboard != null) {
+            try {
+                final ClipData clip = clipboard.getPrimaryClip();
+                if (clip != null && clip.getItemCount() > 0) {
+                    final CharSequence text = clip.getItemAt(0).getText();
+                    return text != null ? text.toString() : null;
+                }
+            } catch (NullPointerException e) {
+                // may happen if clipboard service throws exception while reading clipboard contents
```

Comment:
Can't we avoid the NPE here without using try/catch?

Reply:
There seem to be two possible causes for errors:

The "clipboard != null" check is added to catch if `getSystemService()` returns null, similar check as in `copyToClipboard()` a few lines further up.

The NPE does not seem to get triggered directly by our code. From looking at the log (see linked issue) it seems to be a "bit further down the pipe" during unparceling, that's why I added the try/catch. But maybe there's a way to check this upfront?

Reply:
`ClipboardManager` seems to have a method `hasPrimaryClip`. Maybe checking this one first prevents the NPE? But this is just a guess, we would need to look into the (Android) source code (which is easy in AS)

Reply:
This is the part being called in ClipboardManager:

```
public @Nullable ClipData getPrimaryClip() {
    try {
        return mService.getPrimaryClip(mContext.getOpPackageName(), mContext.getUserId());
    } catch (RemoteException e) {
        throw e.rethrowFromSystemServer();
    }
}
```

Thanks for the pointer to `hasPrimaryClip`. I can change the PR to use this instead of the try/catch. 
I cannot say what's the actual cause for the error I found in the logs. Maybe it's device-specific (configuration, old Android version etc.) and not worth doing a try/catch for a rare occurrence.

Reply:
Updated PR and its title accordingly